src/snowflake/cli/plugins/nativeapp/artifacts.py

```diff
@@ -140,34 +241,22 @@ def _add(self, src: Path, dest: Path, map_as_child: bool) -> None:
             )
 
         if self._deploy_root in absolute_src.parents:
-            # ignore this item since it's in the deploy root. This can happen if the bundle map is create
+            # ignore this item since it's in the deploy root. This can happen if the bundle map is created
             # after the bundle step and a project is using rules that are not sufficiently constrained.
             # Since the bundle step starts with deleting the deploy root, we wouldn't normally encounter this situation.
             return
 
         canonical_src = self._canonical_src(src)
         canonical_dest = self._canonical_dest(dest)
 
-        src_is_dir = absolute_src.is_dir()
         if map_as_child:
             # Make sure the destination is a child of the original, since this was requested
             canonical_dest = canonical_dest / canonical_src.name
-            dest_is_dir = src.is_dir()
-
-        # Verify that multiple files are not being mapped to a single file destination
-        current_sources = self._dest_to_src.setdefault(canonical_dest, [])
-        if not dest_is_dir:
-            # the destination is a file
-            if (canonical_src not in current_sources) and len(current_sources) > 0:
-                raise TooManyFilesError(dest)
+            dest_is_dir = absolute_src.is_dir()
```

Comment:
To clarify, we are ruling out if a user provided something like this?
```
src: some/dir
dest: path/to/file
```
This is a mismatch and should throw an error?

Reply:
this particular line is a bug fix, it only worked because project dir == cwd before. What this logic does is deal with src -> dest vs src -> dest/. In the first case, we're renaming src to dest in the deploy root. In the second case, we're creating dest/src in the deploy root. So this particular condition is updating the dest to add that level of nesting.

Reply:
"project dir == cwd" assumption being key, got it.